### PR TITLE
Support stable Python3.9.

### PR DESCRIPTION
**Pull request by hadialqattan**
Fix: #1747

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -28,9 +28,7 @@ matrix:
     - name: "3.8"
       python: 3.8
     - name: "3.9"
-      python: 3.9-dev
-  allow_failures:
-    - python: 3.9-dev
+      python: 3.9
```
Comment (hugovk) [inline .travis.yml:31]: Travis CI doesn't yet have `3.9` so we need to stick with `3.9-dev` for now.

Here's the tracking issue to +1:

* https://travis-ci.community/t/python-3-9-0-build/10091?u=hugovk

**Comment by hugovk** [comment]
Something extra: the usage in the README (plus the autogenerated `docs/installation_and_usage.md`) can be updated by replacing it with output of `black --help` on master. There's now a `py39` in there.

**Comment by hadialqattan** [comment]
> Something extra: the usage in the README (plus the autogenerated `docs/installation_and_usage.md`) can be updated by replacing it with output of `black --help` on master. There's now a `py39` in there.

A good catch!

**Comment by hadialqattan** [comment]
Here we have an issue with fuzz workflow:
```log
2020-10-08T14:09:27.5958926Z ##[group]Run coverage run fuzz.py
2020-10-08T14:09:27.5959549Z [36;1mcoverage run fuzz.py[0m
2020-10-08T14:09:27.5959957Z [36;1mcoverage report[0m
2020-10-08T14:09:27.6010681Z shell: /bin/bash -e {0}
2020-10-08T14:09:27.6011016Z env:
2020-10-08T14:09:27.6011463Z   pythonLocation: /opt/hostedtoolcache/Python/3.9.0/x64
2020-10-08T14:09:27.6011987Z ##[endgroup]
2020-10-08T14:09:47.8690169Z <string>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
2020-10-08T14:09:47.8692080Z <string>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
2020-10-08T14:09:47.8701479Z <string>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
2020-10-08T14:10:08.4345243Z   File "<string>", line 1
2020-10-08T14:10:08.4480315Z SyntaxError: unexpected character after line continuation character
2020-10-08T14:10:08.4480822Z 
2020-10-08T14:10:08.4481304Z The above exception was the direct cause of the following exception:
2020-10-08T14:10:08.4481703Z 
2020-10-08T14:10:08.4482182Z Traceback (most recent call last):
2020-10-08T14:10:08.4484636Z   File "/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/hypothesmith/syntactic.py", line 110, in draw_symbol
2020-10-08T14:10:08.4485686Z     compile(
2020-10-08T14:10:08.4486472Z SystemError: <built-in function normalize> returned a result with an error set
2020-10-08T14:10:08.4486922Z 
2020-10-08T14:10:08.4487358Z The above exception was the direct cause of the following exception:
2020-10-08T14:10:08.4487755Z 
2020-10-08T14:10:08.4488077Z Traceback (most recent call last):
2020-10-08T14:10:08.4488636Z   File "/home/runner/work/black/black/fuzz.py", line 59, in <module>
2020-10-08T14:10:08.4489219Z     test_idempotent_any_syntatically_valid_python()
2020-10-08T14:10:08.4490030Z   File "/home/runner/work/black/black/fuzz.py", line 17, in test_idempotent_any_syntatically_valid_python
2020-10-08T14:10:08.4490802Z     max_examples=1000,  # roughly 1k tests/minute, or half that under coverage
2020-10-08T14:10:08.4491959Z   File "/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/hypothesis/core.py", line 1162, in wrapped_test
2020-10-08T14:10:08.4532107Z     raise the_error_hypothesis_found
2020-10-08T14:10:08.4533555Z   File "/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/hypothesmith/syntactic.py", line 89, in do_draw
2020-10-08T14:10:08.4534329Z     result = super().do_draw(data)
2020-10-08T14:10:08.4536372Z   File "/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/hypothesis/extra/lark.py", line 146, in do_draw
2020-10-08T14:10:08.4537140Z     self.draw_symbol(data, start, state)
2020-10-08T14:10:08.4539320Z   File "/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/hypothesmith/syntactic.py", line 107, in draw_symbol
2020-10-08T14:10:08.4540796Z     super().draw_symbol(data, symbol, draw_state)
2020-10-08T14:10:08.4541909Z   File "/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/hypothesis/extra/lark.py", line 174, in draw_symbol
2020-10-08T14:10:08.4542695Z     self.draw_symbol(data, e, draw_state)
2020-10-08T14:10:08.4543797Z   File "/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/hypothesmith/syntactic.py", line 107, in draw_symbol
2020-10-08T14:10:08.4544899Z     super().draw_symbol(data, symbol, draw_state)
2020-10-08T14:10:08.4545993Z   File "/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/hypothesis/extra/lark.py", line 174, in draw_symbol
2020-10-08T14:10:08.4547200Z     self.draw_symbol(data, e, draw_state)
2020-10-08T14:10:08.4548777Z   File "/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/hypothesmith/syntactic.py", line 107, in draw_symbol
2020-10-08T14:10:08.4549599Z     super().draw_symbol(data, symbol, draw_state)
2020-10-08T14:10:08.4550688Z   File "/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/hypothesis/extra/lark.py", line 174, in draw_symbol
2020-10-08T14:10:08.4551459Z     self.draw_symbol(data, e, draw_state)
2020-10-08T14:10:08.4553867Z   File "/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/hypothesmith/syntactic.py", line 107, in draw_symbol
2020-10-08T14:10:08.4554706Z     super().draw_symbol(data, symbol, draw_state)
2020-10-08T14:10:08.4555811Z   File "/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/hypothesis/extra/lark.py", line 174, in draw_symbol
2020-10-08T14:10:08.4556861Z     self.draw_symbol(data, e, draw_state)
2020-10-08T14:10:08.4557977Z   File "/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/hypothesmith/syntactic.py", line 107, in draw_symbol
2020-10-08T14:10:08.4558799Z     super().draw_symbol(data, symbol, draw_state)
2020-10-08T14:10:08.4559871Z   File "/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/hypothesis/extra/lark.py", line 174, in draw_symbol
2020-10-08T14:10:08.4560643Z     self.draw_symbol(data, e, draw_state)
2020-10-08T14:10:08.4561734Z   File "/opt/hostedtoolcache/Python/3.9.0/x64/lib/python3.9/site-packages/hypothesmith/syntactic.py", line 119, in draw_symbol
2020-10-08T14:10:08.4562495Z     raise Exception(
2020-10-08T14:10:08.4563815Z Exception: unexpected error while attempting to compile 'pass;importūą.𤳪ŋ냤ŏᩘGŌı\t\t\x0c\t.ĵ.춠𭝙scÈ  \x0c\x0c\x0c\t\x0c\t\t  \x0c \x0c\x0c\t\t \x0c \t\t\t\t \t\x0c\t   \t \\\t\x0c\x0c\t\r\n,slĢŢ.Ĥĸok\\ \t \t\t\n\x0c\x0c\t\x0c\t.ŔĿ,MwŉÀ\\\t\x0c\x0c\n#Ö\x90¤' in mode=single
```

**Comment by hadialqattan** [comment]
For now, I'm gonna remove 3.9 from the `fuzz` workflow, but later I think we should add it.

**Comment by ichard26** [comment]
We should probably hold off declaring Python 3.9 support until one of our dependencies, `typed_ast`, has wheels for Windows + Python 3.9. Needing Microsoft Visual C++ 14.0 or higher doesn’t make for a good install experience :-) 

It’s tracked through https://github.com/python/typed_ast/issues/146 (via #1744).

**Review by JelleZijlstra — APPROVED** [review]

**Comment by hadialqattan** [comment]
> We should probably hold off declaring Python 3.9 support until one of our dependencies, `typed_ast`, has wheels for Windows + Python 3.9. Needing Microsoft Visual C++ 14.0 or higher doesn’t make for a good install experience :-)
> 
> It’s tracked through [python/typed_ast#146](https://github.com/python/typed_ast/issues/146) (via #1744).

I see your point, but actually, almost every Python package has a somehow issue with Windows 😇!

So I think we can document and test for Python3.9 even if we haven't fixed the `typed_ast` Windows issue.

**Comment by cooperlees** [comment]
I see both sides here. I feel if we document the known issues (typed ast) we do nothing but gain tested and documented support for 3.9. We can then just do the PR to cleanup the documentation etc. once that issue is closed. So I'm a +1 on merging and thanks @hadialqattan for the work here!

**Comment by hadialqattan** [comment]
> I see both sides here. I feel if we document the known issues (typed ast) we do nothing but gain tested and documented support for 3.9. We can then just do the PR to cleanup the documentation etc. once that issue is closed. So I'm a +1 on merging and thanks @hadialqattan for the work here!

That's my pleasure.

**Pull request merged**